js/Scene3D.js

```diff
@@ -1,399 +1,408 @@
+                    this._recolor();
+                    this._notify(Scene3D.Events.CHANGE);
+                }
+            }
         },
 
-        set: function(value) {
-            this._colorMap = value;
-            if (this._mesh && this._spots && this._mapping) {
-                this._recolor();
-                this._notify(Scene3D.Events.CHANGE);
+        position: {
+            get: function () {
+                return this._scene.position.clone();
             }
-        }
-    },
+        },
 
-    position: {
-        get: function() {
-            return this._scene.position.clone();
-        }
-    },
+        render: {
+            value: function (renderer, camera) {
+                this._frontLight.position.set(camera.position.x, camera.position.y, camera.position.z);
+                renderer.render(this._scene, camera);
+            }
+        },
 
-    render: {
-        value: function(renderer, camera) {
-            this._frontLight.position.set(camera.position.x, camera.position.y, camera.position.z);
-            renderer.render(this._scene, camera);
-        }
-    },
-
-    raycast: {
-        value: function(raycaster) {
-            if (!this._mesh || !this._spots || !this._mapping) return null;
-            var message = {
-                positions: this._mesh.geometry.attributes.position.array,
-                origin: new THREE.Vector3().copy(raycaster.ray.origin),
-                direction: new THREE.Vector3().copy(raycaster.ray.direction),
-                matrixWorld: new THREE.Matrix4().copy(this._mesh.matrixWorld),
-            };
-            var closestSpotIndeces = this._mapping.closestSpotIndeces;
-            var spots = this._spots;
-            var worker = new Worker('js/workers/Raycaster.js');
-
-            var promise = new Promise(function(accept, reject) {
-                worker.onmessage = function(event) {
-                    worker.terminate();
-                    var face = event.data;
-                    var spotIndex = -1;
-                    for (var i in (face || {})) {
-                        if (closestSpotIndeces[face[i]] >= 0) {
-                            spotIndex = closestSpotIndeces[face[i]];
-                            break;
+        raycast: {
+            value: function (raycaster) {
+                if (!this._mesh || !this._spots || !this._mapping) return null;
+                var message = {
+                    positions: this._mesh.geometry.attributes.position.array,
+                    origin: new THREE.Vector3().copy(raycaster.ray.origin),
+                    direction: new THREE.Vector3().copy(raycaster.ray.direction),
+                    matrixWorld: new THREE.Matrix4().copy(this._mesh.matrixWorld),
+                };
+                var closestSpotIndeces = this._mapping.closestSpotIndeces;
+                var spots = this._spots;
+                var worker = new Worker('js/workers/Raycaster.js');
```

Comment:
:+1:

On (Apr-15-16|14:44), Ivan Protsyuk wrote:
> > -                    for (var i in (face || {})) {
> > -                        if (closestSpotIndeces[face[i]] >= 0) {
> > -                            spotIndex = closestSpotIndeces[face[i]];
> > -                            break;
> > +        raycast: {
> > +            value: function (raycaster) {
> > +                if (!this._mesh || !this._spots || !this._mapping) return null;
> > +                var message = {
> > +                    positions: this._mesh.geometry.attributes.position.array,
> > +                    origin: new THREE.Vector3().copy(raycaster.ray.origin),
> > +                    direction: new THREE.Vector3().copy(raycaster.ray.direction),
> > +                    matrixWorld: new THREE.Matrix4().copy(this._mesh.matrixWorld),
> > +                };
> > +                var closestSpotIndeces = this._mapping.closestSpotIndeces;
> > +                var spots = this._spots;
> > +                var worker = new Worker('js/workers/Raycaster.js');
> 
> @ElDeveloper, ok, let's do as per that example from require.js unit-test, and see if it works :)
> 
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/ili-toolbox/ili/pull/67/files/4f53e58231f03b3efc46368848547a7d50402d81#r59945639